app/scripts/lib/metaRPCClientFactory.js

```diff
@@ -17,34 +18,46 @@ class MetaRPCClient {
     });
   }
 
+  onUncaughtError(handler) {
+    this.uncaughtErrorChannel.addListener('error', (error) => {
+      handler(error);
+    });
+  }
```

Comment:
We should use this new method, probably by adding a listener next to [this](https://github.com/MetaMask/metamask-extension/blob/09d81ac/ui/index.js/#L131-L141) in `ui/index.js`.

That should be done in a follow-up, though.